src/handlers/testing/align/util/emitter.ts

```diff
@@ -0,0 +1,49 @@
+import { EventEmitter } from 'events';
+import { z } from 'zod';
+import type { HandlerOf } from '../../../../util/types';
+import { zTestCaseResult } from './models';
+
+export enum EventName {
+  TEST_CASE_RESULT = 'TEST_CASE_RESULT',
+}
+
+const eventSchemas = {
+  [EventName.TEST_CASE_RESULT]: zTestCaseResult,
+};
+
+export type EventSchemas = {
+  [K in EventName]: z.infer<(typeof eventSchemas)[K]>;
+};
+
+class TypedEventEmitter {
```

Comment:
should pull this out into a shared util class, need to pass EventSchemas as a generic